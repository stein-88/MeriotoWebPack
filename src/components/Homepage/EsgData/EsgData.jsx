import React from 'react'
import schema from './data'

const EsgData = () => {
    const { titulo, text } = schema
    return (
        <div className="row">
            <div className="col-12 text-muted">
                {titulo && <h2 className="h1 pt-5">{titulo}</h2>}
                {text && text.length > 0 && text.map((ele, i) => (
                    <p key={`esg ${i + 1}`} className="h3 my-4">{ele}</p>
                ))}
            </div>
        </div>
    )
}

export default EsgData
