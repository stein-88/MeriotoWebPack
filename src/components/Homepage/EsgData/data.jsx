const schema = {
    titulo: 'Globechain the ESG data reuse solution making waste a resource for everyone',
    text: [
        'We are the leading ESG data reuse marketplace connecting enterprises with non profits, small businesses and people to redistribute unneeded items.',
        'We generate ESG data on your impact, help you reduce waste costs and achieve your sustainability and SDG targets.',
    ],
}

export default schema
