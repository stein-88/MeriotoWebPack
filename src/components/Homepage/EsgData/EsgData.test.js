import React from 'react'
import renderer from 'react-test-renderer'
import EsgData from './EsgData'

describe('testing EsgData', () => {
    const compEsgData = renderer.create(<EsgData />)
    test('Should render without props', () => {
        const jsonEsgData = compEsgData.toJSON()
        expect(jsonEsgData).toMatchSnapshot()
    })
    test('Should have 1 title and 2 cards', () => {
        const resultTitle = compEsgData.root.findAllByProps({ className: 'h1 pt-5' })
        const resultTxt = compEsgData.root.findAllByProps({ className: 'h3 my-4' })
        expect(resultTxt).toHaveLength(2)
        expect(resultTitle).toHaveLength(1)
    })
})