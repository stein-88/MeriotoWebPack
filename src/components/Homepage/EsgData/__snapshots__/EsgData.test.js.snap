// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing EsgData Should render without props 1`] = `
<div
  className="row"
>
  <div
    className="col-12 text-muted"
  >
    <h2
      className="h1 pt-5"
    >
      Globechain the ESG data reuse solution making waste a resource for everyone
    </h2>
    <p
      className="h3 my-4"
    >
      We are the leading ESG data reuse marketplace connecting enterprises with non profits, small businesses and people to redistribute unneeded items.
    </p>
    <p
      className="h3 my-4"
    >
      We generate ESG data on your impact, help you reduce waste costs and achieve your sustainability and SDG targets.
    </p>
  </div>
</div>
`;
